File: src/main/java/io/spring/springbatch/HelloJobConfiguration.java
package io.spring.springbatch;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.batch.core.Job;
import org.springframework.batch.core.Step;
import org.springframework.batch.core.configuration.annotation.JobBuilderFactory;
import org.springframework.batch.core.configuration.annotation.StepBuilderFactory;
import org.springframework.batch.repeat.RepeatStatus;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Slf4j
@Configuration
@RequiredArgsConstructor
public class HelloJobConfiguration {
    private final JobBuilderFactory jobBuilderFactory;
    private final StepBuilderFactory stepBuilderFactory;

    @Bean
    public Job helloJob() {
        return this.jobBuilderFactory.get("helloJob")
                .start(helloFirstStep())
                .next(helloSecondStep())
                .build();
    }

    @Bean
    public Step helloSecondStep() {
        return this.stepBuilderFactory.get("helloFirstStep")
                .tasklet((contribution, chunkContext) -> {
                    log.info("====================");
                    log.info("Hello, Spring Batch!");
                    log.info("====================");
                    return RepeatStatus.FINISHED;
                })
                .build();
    }

    @Bean
    public Step helloFirstStep() {
        return this.stepBuilderFactory.get("helloSecondStep")
                .tasklet((contribution, chunkContext) -> {
                    log.info("=========================");
                    log.info("Second Step was executed!");
                    log.info("=========================");
                    return RepeatStatus.FINISHED;
                })
                .build();
    }
}

File: src/main/kotlin/com/spring/batch/hello/HelloJobConfiguration.kt
package com.spring.batch.hello

import io.klogging.NoCoLogging
import org.springframework.batch.core.Job
import org.springframework.batch.core.Step
import org.springframework.batch.core.job.builder.JobBuilder
import org.springframework.batch.core.repository.JobRepository
import org.springframework.batch.core.step.builder.StepBuilder
import org.springframework.batch.repeat.RepeatStatus
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.transaction.PlatformTransactionManager

@Configuration
class HelloJobConfiguration(
    private val jobRepository: JobRepository,
    private val platformTransactionManager: PlatformTransactionManager,
) : NoCoLogging {
    @Bean
    fun helloJob(): Job {
        return JobBuilder("helloJob", this.jobRepository)
            .start(helloStep())
            .next(helloStep2())
            .build()
    }

    @Bean
    fun helloStep(): Step {
        return StepBuilder("helloStep", this.jobRepository)
            .tasklet({ _, _ ->
                logger.info { "Hello, World!" }
                RepeatStatus.FINISHED
            }, this.platformTransactionManager)
            .build()
    }

    @Bean
    fun helloStep2(): Step {
        return StepBuilder("helloStep2", this.jobRepository)
            .tasklet({ _, _ ->
                logger.info { "Step 2 was executed!" }
                RepeatStatus.FINISHED
            }, this.platformTransactionManager)
            .build()
    }
}
